### PR TITLE
Change chibios_hwdef.py to pre-process defaults files

### PR DESCRIPTION
**Pull request by peterbarker**
Reads the old filepath in, writes the pre-processed result into the builds directory as `preprocessed_defaults.py`.

The only pre-processing directive understood is `@include FILEPATH`.

This PR also changes `KakuteH7Mini-Nand` to not have a copy of `KakuteH7Mini`'s defaults.parm, but rather include `KakuteH7Mini`'s.  It's build is byte-for-byte identical afterwards.

I've also included a file from /tmp/ into a CUAVv5 build and ensured that the default in there appears after flashing.

```
Board               AP_Periph  blimp  bootloader  copter  heli  iofirmware  plane  rover  sub
Durandal                       *      *           *       *                 *      *      *
HerePro             *                                                                     
Hitec-Airspeed      *                                                                     
KakuteH7-bdshot                *      *           *       *                 *      *      *
KakuteH7Mini-Nand              *      *           *       *                 *      *      *
MatekF405                      *      *           *       *                 *      *      *
Pixhawk1-1M-bdshot             *                  *       *                 *      *      *
f103-QiotekPeriph   *                                                                     
f303-Universal      *                                                                     
iomcu                                                           *                         
revo-mini                      *      *           *       *                 *      *      *
skyviper-v2450                                    *                                       
```
(note the `KakuteH7Mini-Nand` in the above table)

This work sponsored by Harris Aerial

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py [hunk 1]
```diff
@@ -2736,24 +2737,60 @@ def build_peripheral_list(self):
...
+        the contents of the so-references file - recursively.'''
+        if depth > 10:
+            raise Exception("include loop")
+        ret = ""
+        with open(defaults_filepath, 'r') as defaults_fh:
+            while True:
+                line = defaults_fh.readline()
+                if line == "":
+                    break
+                m = re.match("^\s*@include\s*([^\s]+)", line)
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py:2751]: why allow white space at start of the line? I think it should be hard against the margin

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 import pickle
 import re
 import shutil
+import tempfile
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py:15]: not used?

**Review by tridge — CHANGES_REQUESTED** [review]
please test with build_binaries or test with out of tree build

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py:15]: Fixed

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py]: Changed

**Review by andyp1per — APPROVED** [review]
I like this change - presumably the testing tridge asked for was done?

**Comment by peterbarker** [comment]
> I like this change - presumably the testing tridge asked for was done?

Yes.

**Pull request merged**